core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -277,8 +277,7 @@ object SparkEnv extends Logging {
       "spark.serializer", "org.apache.spark.serializer.JavaSerializer")
     logDebug(s"Using serializer: ${serializer.getClass}")
 
-    val closureSerializer = instantiateClassFromConf[Serializer](
-      "spark.closure.serializer", "org.apache.spark.serializer.JavaSerializer")
+    val closureSerializer = new JavaSerializer(conf)
```

Comment:
alternatively we could make this a field in `SparkEnv` instead of a constructor argument, so much of the existing code doesn't need to change.